pkg/cluster/config/v1alpha2/types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kind/pkg/kustomize"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Node contains settings for a node in the `kind` Config.
+// A node in kind config represent a container that will be provisioned with all the components
+// required for the assigned role in the Kubernetes cluster
+type Node struct {
+	// TypeMeta representing the type of the object and its API schema version.
+	metav1.TypeMeta `json:",inline"`
+
+	// Replicas is the number of desired node replicas.
+	// Defaults to 1
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Role defines the role of the nodw in the in the Kubernetes cluster managed by `kind`
+	// Defaults to "control-plane"
+	Role NodeRole `json:"role,omitempty"`
+	// Image is the node image to use when running the cluster
+	// TODO(bentheelder): split this into image and tag?
+	Image string `json:"image,omitempty"`
+	// KubeadmConfigPatches are applied to the generated kubeadm config as
+	// strategic merge patches to `kustomize build` internally
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/strategic-merge-patch.md
+	// This should be an inline yaml blob-string
+	KubeadmConfigPatches []string `json:"kubeadmConfigPatches,omitempty"`
+	// KubeadmConfigPatchesJSON6902 are applied to the generated kubeadm config
+	// as patchesJson6902 to `kustomize build`
+	KubeadmConfigPatchesJSON6902 []kustomize.PatchJSON6902 `json:"kubeadmConfigPatchesJson6902,omitempty"`
```

Comment:
side note: i'd be interested to hear user feedback on `KubeadmConfigPatches` and `KubeadmConfigPatchesJSON6902`

Reply:
as possibly the sole user currently:
- it works ish
- sometimes for advanced field patching the 6902 format is more powerful, but the plain yaml patch style is much nicer to write and very simple to understand, it just merges the yaml kinda like apply.
- it's not the _best_ UX, but you also can ignore it and isntead run config patching in a lifecycle hook, though the UX will be worse for that.